src/parser.rs

```diff
@@ -2123,11 +2135,19 @@ impl<'input, 'abbrev, 'unit, Endian> EntriesCursor<'input, 'abbrev, 'unit, Endia
         }
     }
 
+    /// Return the current offset in bytes inside the compilation unit
+    pub fn current_offset(&self) -> usize {
+        let ptr = self.input.as_ptr() as *const u8 as usize;
+        let start_ptr = self.unit.entries_buf.as_ptr() as *const u8 as usize;
+        ptr - start_ptr + self.unit.header_size()
+    }
+
     /// Move the cursor to the next DIE in the tree.
     ///
     /// Returns `Some` if there is a next entry, even if this entry is null.
     /// If there is no next entry, then `None` is returned.
     pub fn next_entry(&mut self) -> ParseResult<Option<()>> {
+        let offset = self.current_offset();
```

Comment:
Move this down to about line 2158. The offset needs to be calculated after we call `after_entry`. The problem is that we lazily parse the attributes for the previous entry, so we need to finish parsing them first. It doesn't affect the dwarfdump example because it displays every attribute, but that won't always be the case.